src/impls.rs

```diff
@@ -177,6 +177,30 @@ where
 	}
 }
 
+impl<K, V> HasTypeId for BTreeMap<K, V>
+where
+	K: Metadata + 'static,
+	V: Metadata + 'static,
+{
+	fn type_id() -> TypeId {
+		TypeIdCustom::new("BTreeMap", Namespace::prelude(), tuple_meta_type!(K, V)).into()
+	}
+}
+
+impl<K, V> HasTypeDef for BTreeMap<K, V>
+where
+	K: Metadata + 'static,
+	V: Metadata + 'static,
+{
+	fn type_def() -> TypeDef {
+		TypeDefEnum::new(vec![
+			EnumVariantTupleStruct::new("Key", vec![UnnamedField::of::<K>()]).into(),
+			EnumVariantTupleStruct::new("Value", vec![UnnamedField::of::<V>()]).into(),
+		])
+		.into()
```

Comment:
A `BTreeMap` is not a Rust `enum` but a Rust `struct` with some elements. Since we are not interested in the actual underlying structures that make up a `BTreeMap` but more in its SCALE encoding please inspect what its SCALE encoding does (probably encodes it as `Vec<(K, V)>` and use that.

Reply:
Thanks for the review. I did not notice that `TypeDefEnum` is wrong for `BTreeMap` as it did work for me :(, maybe the encoding is same accidently under the hood. I'll take a closer look.

Reply:
@Robbepop I have updated based on the `Vec<T>` impl, but still unsure it's right.